api/views/diagnostic.py

```diff
@@ -232,11 +232,27 @@ def _update_or_create_canteen_with_diagnostic(self, row, siret):
             except Exception as e:
                 raise ValidationError({"value_label_hve": number_error_message})
 
+        silently_added_manager_emails = []
+        import_source = "Import massif"
+        if len(row) > 18:
```

Comment:
À discuter si on passe cette validation dans la fonction `_verify_file_size`, ce qui éviterait de faire le traitement au dessus. 

Ça permettrait de rendre la verification de la taille de colonnes prioritaire par rapport aux autres validations. En ayant la validation ici, je pourrais avoir des erreurs type "Champ bio is not a number" et trucs comme ça, et après les avoir corrigés obtenir un "mauvais nombre de colonnes". Idéalement on ne trairait même pas le fichier s'il y a plus de 18 colonnes alors que la personne n'est pas staff.
```
if len(row) > 18 and not self.request.user.is_staff:
                raise PermissionDenied(detail=f"Format fichier : 15-18 ou 11 colonnes attendues, {len(row)} trouvés.")
```

Reply:
Bien vu, mais je ne l'ai pas mis dans _verify_file_size car c'est appelé avant la déconstruction d'une ligne en colonnes